sentinel/src/lib/messages.js

```diff
@@ -85,21 +89,29 @@ module.exports = {
       if (!from) {
         return true;
       }
+
       if (module.exports.isMessageFromGateway(from)) {
         return false;
       }
-      const conf = config.get('outgoing_deny_list');
-      if (!conf) {
-        return true;
-      }
-      return _.every(conf.split(','), s => {
-        // ignore falsey inputs
-        if (!s) {
-          return true;
-        }
-        // return false if we get a case insensitive starts with match
-        return from.toLowerCase().indexOf(s.trim().toLowerCase()) !== 0;
-      });
+
+      const outgoingDenyList = config.get('outgoing_deny_list');
+      const outgoingDenyShorts = config.get('outgoing_deny_shorter_than');
+      const outgoingDenyAlphas = config.get('outgoing_deny_with_alphas');
+      const sanitize = x => x && x.toLowerCase().trim();
+      const isNumeric = x => !isNaN(parseFloat(x)) && isFinite(x);
```

Comment:
Sanitize `from` here so you don't have to do it multiple times below, eg: `from = sanitize(from);`